sound/soc/sof/intel/hda-dsp.c

```diff
@@ -282,6 +282,54 @@ void hda_dsp_ipc_int_disable(struct snd_sof_dev *sdev)
 			HDA_DSP_REG_HIPCCTL_BUSY | HDA_DSP_REG_HIPCCTL_DONE, 0);
 }
 
+static int hda_dsp_wait_d0i3c_done(struct snd_sof_dev *sdev, int retry)
+{
+	struct hdac_bus *bus = sof_to_bus(sdev);
+	u8 reg;
+
+	reg = snd_hdac_chip_readb(bus, VS_D0I3C);
+	while ((snd_hdac_chip_readb(bus, VS_D0I3C) & SOF_HDA_VS_D0I3C_CIP) &&
+	       retry--)
```

Comment:
can you try to make this cleaner?
It's really ugly to read. Don't we have a helper anyways for read with timeout?